### PR TITLE
"Don't directly throw Exception" System.Security.AccessControl

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Contributes to #23748 .
cc: @bartonjs, @ianhays

### CODE REVIEW

File: src/System.Security.AccessControl/src/System/Security/AccessControl/SecurityDescriptor.cs [hunk 1]
```diff
@@ -669,10 +669,8 @@ private static byte[] BinaryFormFromSddlForm(string sddlForm)
 nameof(sddlForm));
                     }
                     else if (error != Interop.Errors.ERROR_SUCCESS)
-                    {
-                        Debug.Assert(false, string.Format(CultureInfo.InvariantCulture, "Unexpected error out of Win32.ConvertStringSdToSd: {0}", error));
```
Comment (bartonjs) [inline src/System.Security.AccessControl/src/System/Security/AccessControl/SecurityDescriptor.cs]: The assert should remain, it's present in .NET Framework: http://referencesource.microsoft.com/#mscorlib/system/security/accesscontrol/securitydescriptor.cs,678

File: src/System.Security.AccessControl/src/System/Security/AccessControl/SecurityDescriptor.cs [hunk 2]
```diff
@@ -669,10 +669,8 @@ private static byte[] BinaryFormFromSddlForm(string sddlForm)
...
-                        // TODO : This should be a Win32Exception once that type is available
-                        throw new Exception();
+                    {                        
+                        throw new Win32Exception(error,SR.Format(SR.AccessControl_UnexpectedError,error));
```
Comment (bartonjs) [inline src/System.Security.AccessControl/src/System/Security/AccessControl/SecurityDescriptor.cs]: Style: There should be spaces after the commas.

File: src/System.Security.AccessControl/src/System/Security/AccessControl/SecurityDescriptor.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/System.Security.AccessControl/src/System/Security/AccessControl/SecurityDescriptor.cs]: done

File: src/System.Security.AccessControl/src/System/Security/AccessControl/SecurityDescriptor.cs (hunk 2 above)
Comment (MarcoRossignoli) [reply src/System.Security.AccessControl/src/System/Security/AccessControl/SecurityDescriptor.cs]: done

**Pull request merged**